DBAccess/MsAccess.cs

```diff
@@ -1,36 +1,96 @@
 ﻿using System;
-using System.Collections.Generic;
+using System.Data.OleDb;
 using System.Data;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Configuration;
+using System.Collections.Generic;
 
 namespace DBAccess
 {
-    class MsAccess : DBAccess
+   public class MsAccess : DBAccess
     {
+        private DataSet set;
+        private OleDbConnection con;
         public MsAccess(string connectionString) : base(connectionString)
         {
-
+            set = new DataSet();
+            try
+            {
+                this.con = new OleDbConnection(connectionString);
+            }
+            catch (Exception e)
+            {
+                this.ProcessException(e);
+            }
+           
+            this.Connect();
         }
         public override void Connect()
         {
-            throw new NotImplementedException();
+            this.CleanStatus();
+            if (this.con.State == ConnectionState.Open) return;
+            try
+            {
+                con.Open();
+            }
+            catch (Exception e)
+            {
+                this.ProcessException(e);
+            }    
         }
 
         public override void Disconnect()
         {
-            throw new NotImplementedException();
+            this.CleanStatus();
+            try
+            {
+                con.Close();
+            }
+            catch (Exception ex)
+            {
+                this.ProcessException(ex);
+            }      
         }
 
-        public override DataTable SqlQuery(string sql, IDictionary<string, object> parameters)
+        public override DataTable SqlQuery(string sql, IDictionary<string, Object> parameters)
         {
-            throw new NotImplementedException();
+            this.CleanStatus();
+            DataTable retorno = null;
+            try
+            {
+                OleDbCommand myCmd = new OleDbCommand(sql, con);
+                foreach (var item in parameters)
+                {
+                    myCmd.Parameters.AddWithValue(item.Key, item.Value);
+                }
+                OleDbDataAdapter dat = new OleDbDataAdapter(myCmd);
+                dat.SelectCommand = myCmd;
+                dat.Fill(retorno);
+            }
+            catch (Exception ex)
+            {
+                this.ProcessException(ex);
+            } 
+            return retorno;
         }
 
-        public override void SqlStatement(string sql, IDictionary<string, object> parameters)
+        public override void SqlStatement(string pSql, IDictionary<string, Object> parameters)
         {
-            throw new NotImplementedException();
+            this.CleanStatus();
+            try
+            {
+                OleDbCommand CmdSql = new OleDbCommand(pSql, con);
```

Comment:
lines 60-64 and 81-85 are almost the same. Please create a private method to handle the parameters.